adam/curriculum/pursuit_curriculum.py

```diff
@@ -0,0 +1,118 @@
+"""
+Curricula for Pursuit Learning Algorithm
+In Pursuit, given a set of scenes and labels, the learner hypothesizes a meaning for the label and uses confidence
+metrics to pursue the strongest hypothesis as long as it is supported by the following scenes.
+Paper: The Pursuit of Word Meanings (Stevens et al., 2017)
+"""
+
+from adam.curriculum import ExplicitWithSituationInstanceGroup
+from adam.curriculum.curriculum_utils import (
+    phase1_instances,
+    PHASE1_CHOOSER,
+    Phase1InstanceGroup,
+    standard_object,
+)
+from adam.ontology.phase1_ontology import (
+    BIRD,
+    BOX,
+    GAILA_PHASE_1_ONTOLOGY,
+    PERSON,
+    BALL,
+    CHAIR,
+    TABLE,
+    DOG,
+)
+from adam.situation.templates.phase1_templates import (
+    Phase1SituationTemplate,
+    object_variable,
+    sampled,
+)
+import random
+
+random.seed(0)
+
+
+def make_simple_pursuit_curriculum(
+    *,
+    num_instances: int = 10,
+    num_noise_instances: int = 0,
+    num_objects_in_instance: int = 3,
+) -> Phase1InstanceGroup:
+    """
+    Creates a Pursuit-learning curriculum with for a set of standard objects.
+    Args:
+        num_instances: total number of learning instances for each object to learn
+        num_noise_instances: number of noisy learning instances in each set of learning instances. A noisy instance in
+        a scene where the utterance doesn't match the situation and perception (e.g. hearing "ball" while seeing a cup).
+        num_objects_in_instance: number of objects in each instance
+    """
+    if num_noise_instances > num_instances:
+        raise RuntimeError("Cannot have more noise than regular exemplars")
+
+    target_objects = [BALL, CHAIR, PERSON, TABLE, DOG, BIRD, BOX]
+    target_object_variables = [
+        object_variable(target.handle + "-target", target) for target in target_objects
+    ]
+    other_objects = [
```

Comment:
Perhaps call this `noise_object_variables`. Also note that nothing restricts the relationship between noise object variables and target object variables, which could result in a noise instance accidentally being non-noisy by a `noise_object_variable` accidentally selecting the right thing, but that we are not going to worry about that case.